app/scripts/controllers/permissions/loggerMiddleware.js

```diff
@@ -0,0 +1,169 @@
+
+const clone = require('clone')
+const { isValidAddress } = require('ethereumjs-util')
+
+const LOG_LIMIT = 100
+
+/**
+ * Create middleware for logging requests and responses to restricted and
+ * permissions-related methods.
+ */
+module.exports = function createLoggerMiddleware ({
+  walletPrefix, restrictedMethods, store, logStoreKey, historyStoreKey, ignoreMethods,
+}) {
+  return (req, res, next, _end) => {
+    let activityEntry, requestedMethods
+    const { origin, method } = req
+    const isInternal = method.startsWith(walletPrefix)
+    if ((isInternal || restrictedMethods.includes(method)) && !ignoreMethods.includes(method)) {
+      activityEntry = logActivity(req, isInternal)
+      if (method === `${walletPrefix}requestPermissions`) {
+        requestedMethods = getRequestedMethods(req)
+      }
+    } else if (method === 'eth_requestAccounts') {
+      activityEntry = logActivity(req, isInternal)
+      requestedMethods = [ 'eth_accounts' ]
+    } else {
+      return next()
+    }
+
+    next(cb => {
+      const time = Date.now()
+      addResponse(activityEntry, res, time)
+      if (!res.error && requestedMethods) {
+        logHistory(requestedMethods, origin, res.result, time, method === 'eth_requestAccounts')
+      }
+      cb()
+    })
+  }
+
+  function logActivity (request, isInternal) {
+    const activityEntry = {
+      id: request.id,
+      method: request.method,
+      methodType: isInternal ? 'internal' : 'restricted',
+      origin: request.origin,
+      request: cloneObj(request),
+      requestTime: Date.now(),
+      response: null,
+      responseTime: null,
+      success: null,
+    }
+    commitActivity(activityEntry)
+    return activityEntry
+  }
+
+  function addResponse (activityEntry, response, time) {
+    if (!response) {
+      return
+    }
+    activityEntry.response = cloneObj(response)
+    activityEntry.responseTime = time
+    activityEntry.success = !response.error
+  }
+
+  function commitActivity (entry) {
+    const logs = store.getState()[logStoreKey]
+    if (logs.length > LOG_LIMIT - 2) {
+      logs.pop()
```

Comment:
`pop` will remove the _last_ item in the array. This will result in the same 100 logs being stuck in memory.
I guessing what was intended is this:
```suggestion
      logs.shift()
```

Reply:
I made this change in 23c5e562c